code/modules/projectiles/guns/ballistic/shotgun.dm

```diff
@@ -188,6 +190,26 @@
 	. += "You can remove a secondary magazine by alt-right-clicking [src]."
 	. += "Right-click to swap the magazine to the secondary position, and vice versa."
 
+/obj/item/gun/ballistic/shotgun/bulldog/ui_action_click(mob/user, actiontype)
+	if(istype(actiontype, /datum/action/item_action/toggle_firemode))
+		burst_select()
+	else
+		..()
+
+/obj/item/gun/ballistic/shotgun/bulldog/proc/burst_select()
```

Comment:
There's no reason this should be a copypaste of the omonimously named `gun/ballistic/automatic/burst_select()`, or for the original proc to be a children of `gun/ballistic/automatic`, when `burst_size` and `fire_delay` are generic variables that all gun items have.

TL;DR: make a generic `burst_select()` proc for all guns or ballistic guns.

Reply:
I got very lazybrain here even though my brain was also telling me 'just make it a generic proc. I'll get it fixed up.